### ISSUE TITLE
Missing `instances_train.json` in FSCD-147 dataset

### ISSUE THREAD

**Issue by 5JM** [Chipmunk-g4/Template-Matching-and-Regression#6]
Hi,

First of all, thank you for the great work and for releasing the FSCD-147 dataset. It has been very helpful for my research.

While preparing the dataset, I noticed that the file `instances_train.json` seems to be missing from the dataset directory.

From the dataset structure, it appears that this file should exist for the training annotations, but I could not find it in the downloaded dataset.

While searching for related information, I came across a similar discussion in another repository:
https://github.com/VinAIResearch/Counting-DETR/issues/9

However, since that issue does not seem to provide a clear resolution, I wanted to ask here to confirm.

Could you please clarify whether:
1. `instances_train.json` should be included in the dataset, or  
2. the training annotations are provided through another file?

If I misunderstood the dataset structure or missed something during setup, I would really appreciate your guidance.

Thank you again for sharing this valuable resource.

**Comment by YWliu040120** [comment]
> Hi,
> 
> First of all, thank you for the great work and for releasing the FSCD-147 dataset. It has been very helpful for my research.
> 
> While preparing the dataset, I noticed that the file `instances_train.json` seems to be missing from the dataset directory.
> 
> From the dataset structure, it appears that this file should exist for the training annotations, but I could not find it in the downloaded dataset.
> 
> While searching for related information, I came across a similar discussion in another repository: [VinAIResearch/Counting-DETR#9](https://github.com/VinAIResearch/Counting-DETR/issues/9)
> 
> However, since that issue does not seem to provide a clear resolution, I wanted to ask here to confirm.
> 
> Could you please clarify whether:
> 
> 1. `instances_train.json` should be included in the dataset, or
> 2. the training annotations are provided through another file?
> 
> If I misunderstood the dataset structure or missed something during setup, I would really appreciate your guidance.
> 
> Thank you again for sharing this valuable resource.

same as you!

**Comment by YWliu040120** [comment]
<img width="1111" height="152" alt="Image" src="https://github.com/user-attachments/assets/d15c63db-23a3-4b04-acfa-ac93b1bc1e1c" />

**Comment by Chipmunk-g4** [comment]
Hi,

Thank you for your question, and sorry for the delayed response.

You are right that `instances_train.json` is not included in the original FSC-147 dataset.

For a fair comparison with previous methods, we used the pseudo-labels proposed in PseCo (https://arxiv.org/pdf/2311.12386). As described in Section 4.1 of the paper:

"Since there are no box annotations in FSC-147 training data, we generate the pseudo-labels through the ground-truth point prompts. Although the pseudo-labels may be noisy, we find that it is sufficient to train a good classification network."

We used these pseudo-labels as `instances_train.json`.

If needed, you can download the file from the following Google Drive link:
https://drive.google.com/file/d/1mbOgd9KmTi0jX0q38F0TN1Gqb8O0RNc4/view?usp=sharing

**Comment by 5JM** [comment]
Thanks for sharing `instances_train.json`. Really appreciate it!
